### PR TITLE
Disable 1.4 ScheduledJob tests when master >= 1.5

### LINKED ISSUE #38232: [Upgrade test] "ScheduledJob should not emit unexpected warnings" failing in ci-kubernetes-e2e-gce-1.4-1.5-upgrade-master
Test has been failing [100% of the time](https://k8s-testgrid.appspot.com/google-1.4-1.5-upgrade#gce-cvm-1.4-cvm-1.5-upgrade-master&width=20&sort-by-failures=&include-filter-by-regex=ScheduledJob%20should%20not%20emit%20unexpected%20warnings) in [ci-kubernetes-e2e-gce-1.4-1.5-upgrade-master](https://k8s-testgrid.appspot.com/google-1.4-1.5-upgrade#gce-cvm-1.4-cvm-1.5-upgrade-master&width=20&sort-by-failures=). Sample failure: [link](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-1.4-1.5-upgrade-master/15).

Based on [spreadsheet](https://docs.google.com/spreadsheets/d/1sAZqyWE--0fvN1PIuKTw9JwmcXNz6tQIm1MrjdILtm4/edit#gid=1325239137&vpid=A1) tracking 1.5 upgrade test failures created by @krousey.

### PR DESCRIPTION
**Pull request by krousey**
**What this PR does / why we need it**: To skip expected failures in upgrade testing.

**Which issue this PR fixes**: fixes #38232

**Special notes for your reviewer**: This is a direct commit. Not a cherry-pick.

### CODE REVIEW

**Comment by krousey** [comment]
@jessfraz This doesn't belong in the master and 1.5 as it does not apply. So this isn't exactly a cherry-pick. Please let me know if there is a better way to go about this than a direct PR to the 1.4 release.

File: test/e2e/scheduledjob.go [hunk 1]
```diff
@@ -46,6 +47,19 @@ var _ = framework.KubeDescribe("ScheduledJob", func() {
...
+		// can't work with that.
+		v := version.MustParse("1.5.0")
+		gte, err := framework.ServerVersionGTE(v, f.Client)
+		if err != nil {
+			framework.Failf("Failed to get server version: %v", err)
+		}
+		if !gte {
+			framework.Skipf("Not supported for server versions after %q", v)
+		}
+
```
Comment (soltysh) [inline test/e2e/scheduledjob.go]: Why not using `framework.SkipUnlessServerVersionGTE`, like [here](https://github.com/kubernetes/kubernetes/blob/29d10a817ae6eadfd271447a2ab6bbe50856d458/test/e2e/kubectl.go#L262) ?

**Review by soltysh — CHANGES_REQUESTED** [review]

File: test/e2e/scheduledjob.go (hunk 1 above)
Comment (soltysh) [inline test/e2e/scheduledjob.go]: But that's exactly what it does, it checks if the server version is at least X. Though I agree, the naming is unfortunate. Compare [these lines](https://github.com/kubernetes/kubernetes/blob/29d10a817ae6eadfd271447a2ab6bbe50856d458/test/e2e/framework/util.go#L410-L418) with your code.

Comment (soltysh) [inline test/e2e/scheduledjob.go]: Or that 😉

**Review by soltysh — APPROVED** [review]

**Comment by soltysh** [comment]
LGTM

**Comment by soltysh** [comment]
Thank you!

**Comment by krousey** [comment]
@k8s-bot unit test this

**Comment by krousey** [comment]
@k8s-bot unit test this

**Pull request merged**